src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsBuildPlanCreator.java

```diff
@@ -112,19 +114,26 @@ public String getPipelineScript(ProgrammingLanguage programmingLanguage, VcsRepo
         return replacements;
     }
 
-    private String[] getResourcePath(ProgrammingLanguage programmingLanguage, boolean isStaticCodeAnalysisEnabled, boolean isSequentialRuns) {
+    private String[] getResourcePath(ProgrammingLanguage programmingLanguage, Optional<ProjectType> projectType, boolean isStaticCodeAnalysisEnabled, boolean isSequentialRuns) {
         final var pipelineScriptFilename = isStaticCodeAnalysisEnabled ? "Jenkinsfile-staticCodeAnalysis" : "Jenkinsfile";
         final var regularOrSequentialDir = isSequentialRuns ? "sequentialRuns" : "regularRuns";
         final var programmingLanguageName = programmingLanguage.name().toLowerCase();
-        return new String[] { "templates", "jenkins", programmingLanguageName, regularOrSequentialDir, pipelineScriptFilename };
+
+        if (projectType.isPresent() && programmingLanguage.equals(ProgrammingLanguage.C)) {
```

Comment:
It is always a good practice to revert the order in `programmingLanguage.equals(ProgrammingLanguage.C)` and put the enum first, i.e. `ProgrammingLanguage.C.equals(programmingLanguage)` to avoid null pointer exception if programmingLanguage is null but in this case it will break in line 120 `final var programmingLanguageName = programmingLanguage.name().toLowerCase();`
Maybe it is a good idea to add a check for programmingLanguage.

Reply:
Have now reverted the `equals` order and added a Null check for ProgrammingLanguage

Thanks 🙂